apps/backend/src/app/modules/maps/map-submission.service.ts

```diff
@@ -496,6 +496,41 @@ export class MapSubmissionService {
 
     await this.createOrUpdatePrivateTestingInvites(this.db, mapID, requestIDs);
   }
+
+  async testingRequestResponse(mapID: number, userID: number, accept: boolean) {
+    const map = await this.db.mMap.findUnique({
+      where: { id: mapID },
+      include: { testingRequests: true }
+    });
+
+    if (!map) {
+      throw new NotFoundException('Map does not exist');
+    }
+
+    if (map.status !== MapStatusNew.PRIVATE_TESTING) {
+      throw new ForbiddenException('Map is not in private testing');
+    }
```

Comment:
Yeah this is complicated. It really depends how testing requests are shown on the site/ingame, and that's not clear yet since we still haven't figured out how to do a notification system that doesn't suck - I keep meaning to give that more thought.

If it weren't possible for a map to *return* to private testing from statuses further down the map submission pipeline, this would be pretty simple to handle: we just delete all `MapTestingRequest`s for that map and any corresponding notifications. Since it *is* possible to *return* to private testing though, I think we should keep the testing requests, and just remove the notifications, then if a map moves back to private testing, regenerate the notifications for `MapTestingRequestState.UNREAD` requests. We can even artificially give them the `createdAt` date of when the map *first* went into private testing.

Only downside is we store a bit more in DB since we're not frequently removing entries from the `MapTestingRequest` table, but that will barely take up any storage.

If this approach seems okay to you, the code in this PR can be left as-is and we'll just implement this system after we re-do the notification system. In fact, as I'm thinking about it I'm realizing we should really get on and design that, potentially @GordiNoki or something else could work on this as I do the more general map submission stuff. Maybe we should try discuss that next Saturday?

Reply:
Yeah let's leave this for later then, keeping in mind the spam impact a map can have when going between states when we discuss it later.